clients/admin-ui/src/features/common/ConfirmationModal.tsx

```diff
@@ -14,37 +16,61 @@ interface Props {
   isOpen: boolean;
   onClose: () => void;
   onConfirm: () => void;
-  title: string;
-  message: ReactNode;
+  title?: string;
+  message?: ReactNode;
+  cancelButtonText?: string;
+  cancelButtonThemingProps?: ThemingProps<"Button">;
+  continueButtonText?: string;
+  continueButtonThemingProps?: ThemingProps<"Button">;
+  isLoading?: boolean;
+  returnFocusOnClose?: boolean;
+  isCentered?: boolean;
 }
 const ConfirmationModal = ({
   isOpen,
   onClose,
   onConfirm,
   title,
   message,
+  cancelButtonText,
+  cancelButtonThemingProps,
+  continueButtonText,
+  continueButtonThemingProps,
+  isLoading,
+  returnFocusOnClose,
+  isCentered,
 }: Props) => (
-  <Modal isOpen={isOpen} onClose={onClose} size="lg">
+  <Modal
+    isOpen={isOpen}
+    onClose={onClose}
+    size="lg"
+    returnFocusOnClose={returnFocusOnClose || true}
+    isCentered={isCentered}
+  >
     <ModalOverlay />
     <ModalContent textAlign="center" p={2} data-testid="confirmation-modal">
-      <ModalHeader fontWeight="medium">{title}</ModalHeader>
-      <ModalBody>{message}</ModalBody>
+      {title ? <ModalHeader fontWeight="medium">{title}</ModalHeader> : null}
+      {message ? <ModalBody>{message}</ModalBody> : null}
       <ModalFooter>
         <SimpleGrid columns={2} width="100%">
           <Button
             variant="outline"
             mr={3}
             onClick={onClose}
             data-testid="cancel-btn"
+            isDisabled={isLoading}
+            {...cancelButtonThemingProps}
           >
-            Cancel
+            {cancelButtonText || "Cancel"}
```

Comment:
I made all of the current defaults act as fallback values if the text isn't passed in so current modals don't change.